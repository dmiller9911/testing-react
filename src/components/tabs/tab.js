import React from 'react';
import { css } from 'aphrodite';
import { styles } from './tabs.styles';

export const Tab = ({ label, active, onClick, testId }) => (
  <div
    data-testid={testId || label}
    className={css(styles.tab, active && styles.tabActive)}
    onClick={onClick}
  >
    {label}
  </div>
);
