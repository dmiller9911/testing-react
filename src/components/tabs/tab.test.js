import React from 'react';
import { shallow } from 'enzyme';
import { Tab } from './tab';

it('renders inactive', () => {
  const view = shallow(<Tab label="Label" onClick={jest.fn()} />);
  expect(view).toMatchSnapshot();
});

it('renders active', () => {
  const view = shallow(<Tab active label="Label" onClick={jest.fn()} />);
  expect(view).toMatchSnapshot();
});
