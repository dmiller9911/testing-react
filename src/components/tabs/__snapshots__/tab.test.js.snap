// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active 1`] = `
<div
  className="tab_1ccfp04-o_O-tabActive_1c216xt"
  data-testid="Label"
  onClick={[MockFunction]}
>
  Label
</div>
`;

exports[`renders inactive 1`] = `
<div
  className="tab_1ccfp04"
  data-testid="Label"
  onClick={[MockFunction]}
>
  Label
</div>
`;
